model/model.go

```diff
@@ -206,6 +208,89 @@ func (model Model) PrintModel() {
 	model.GetLogger().LogModel(modelInfo)
 }
 
+func (model Model) SortPoliciesBySubjectHierarchy() error {
+	if model["e"]["e"].Value != "subjectPriority(p_eft) || deny" {
+		return nil
+	}
+	subIndex := -1
+	for ptype, assertion := range model["p"] {
+		for index, token := range assertion.Tokens {
+			if token == fmt.Sprintf("%s_sub", ptype) {
+				subIndex = index
+				break
+			}
+		}
+		if subIndex == -1 {
+			continue
+		}
+		policies := assertion.Policy
+		subjectHierarchyMap, err := getSubjectHierarchyMap(model)
+		if err != nil {
+			return err
+		}
+		sort.SliceStable(policies, func(i, j int) bool {
+			p1 := subjectHierarchyMap[policies[i][subIndex]]
+			p2 := subjectHierarchyMap[policies[j][subIndex]]
+			return p1 > p2
+		})
+		for i, policy := range assertion.Policy {
+			assertion.PolicyMap[strings.Join(policy, ",")] = i
+		}
+	}
+	return nil
+}
+
+func getSubjectHierarchyMap(model Model) (map[string]int, error) {
+	policies := model["g"]["g"].Policy
+	subjectHierarchyMap := make(map[string]int)
+	// Tree structure of role
+	policyMap := make(map[string][]string)
+	for _, policy := range policies {
+		if len(policy) < 2 {
+			return nil, errors.New("policy g expect 2 more params")
+		}
+		child := policy[0]
+		parent := policy[1]
+		policyMap[parent] = append(policyMap[parent], child)
+		if _, ok := subjectHierarchyMap[child]; !ok {
+			subjectHierarchyMap[child] = 0
+		}
+		if _, ok := subjectHierarchyMap[parent]; !ok {
+			subjectHierarchyMap[parent] = 0
+		}
+		subjectHierarchyMap[child] = 1
```

Comment:
What if there is a domain?

Reply:
In this issue,the method can't support domain.I'm not sure if there is such a demand

Reply:
I would recommend it